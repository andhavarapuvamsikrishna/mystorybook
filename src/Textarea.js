import React, {Component} from "react"
import textarea from "./textarea.css"
class Textarea extends Component {
    constructor() {
        super()
        this.state = {
            
        }
        this.handleChange = this.handleChange.bind(this)
    }
    
    handleChange(event) {
        const {name, value} = event.target
        
        this.setState({
            [name]: value
        })
    }
    
    render() {
        return (
            <form> 
                <textarea  className="txt" placeholder="enter some text" value={this.handleChange.value} onChange={this.handleChange}/>
            </form>
        )
    }
}

export default Textarea