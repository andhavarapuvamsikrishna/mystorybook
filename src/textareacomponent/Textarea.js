import React, {Component} from "react"
import textarea from "./textarea.css"
class Textarea extends Component {
    constructor() {
        super()
        this.state = {
            characters:40,
            r1:""
        }
        this.handleChange = this.handleChange.bind(this)
    }
    
    handleChange(event) {
        const {name, value} = event.target
        let v=value.length
        let m=40-v
        if(m<0)
        {
            this.setState((prevState) => ({
                [name]: prevState.r1,
                //characters:20-v
            }))  
        }
        else
        {
            this.setState({
                [name]: value,
                characters:40-v
            })
        }
        
    }
    
    render() {
        return (
            <form> 
                <textarea  data-test="texta" className="txt" placeholder="enter some text" name="r1" value={this.state.r1} onChange={this.handleChange}/>
                <br />
        <label>you have </label>
        <input type="text" name="r1" style={{width:"30px"}} value={this.state.characters}  readOnly/>left
            </form>
        )
    }
}

export default Textarea