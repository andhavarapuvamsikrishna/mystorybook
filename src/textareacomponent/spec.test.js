import React from "react"
import { shallow } from "enzyme"
import Textarea from "./Textarea"
import { findByTestAttr } from "../utils/find.js"
import renderer from "react-test-renderer" 
const setUp=(props={}) =>{
    const component =shallow(<Textarea {...props} />)
    return component
    }

describe("my text area component",() =>{
    it("should match the snapshot",() =>{
        const tree=renderer.create(<Textarea />).toJSON()
        expect(tree).toMatchSnapshot() 
  
    })

    it("should renspond to change",()=>{
       
        const wrapper=setUp()
        let textarea=findByTestAttr(wrapper,"texta")
        textarea.simulate('change',{target:{name:"r1",value:"vamsi"}})
        textarea=findByTestAttr(wrapper,"texta")
        expect(textarea.props().value).toEqual("vamsi")    
  
    })
    it("should have maximum 40 characters",()=>{
       //let v="my name is vamsikrishna, iam doing intern"
        const wrapper=setUp()
        let textarea=findByTestAttr(wrapper,"texta")
        textarea.simulate('change',{target:{name:"r1",value:"my name is vamsi krishna iam an internal"}})
        textarea.simulate('change',{target:{name:"r1",value:"my name is vamsi krishna iam an internaladff"}})
        textarea=findByTestAttr(wrapper,"texta")  
        expect(wrapper.state('r1')).toEqual("my name is vamsi krishna iam an internal")      
      
    })      
        
})  
         