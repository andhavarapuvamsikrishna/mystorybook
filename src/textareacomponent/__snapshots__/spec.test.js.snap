// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my text area component should match the snapshot 1`] = `
<form>
  <textarea
    className="txt"
    data-test="texta"
    name="r1"
    onChange={[Function]}
    placeholder="enter some text"
    value=""
  />
  <br />
  <label>
    you have 
  </label>
  <input
    name="r1"
    readOnly={true}
    style={
      Object {
        "width": "30px",
      }
    }
    type="text"
    value={40}
  />
  left
</form>
`;
